firestore.rules

```diff
@@ -63,14 +68,35 @@ service cloud.firestore {
 
     // Used for deletion of user data
     match /deletion/users {
-      function markingCurrentUserDoc() {
-        // Verify that the current user is adding their user ID to the userDocs field
-        return isSignedIn() && request.auth.uid in request.resource.data.userDocs;
+      // Verify that a (1) user is signed in, (2) the only modified field is
+      // userDocs, (3) the userDocs array is increased in size by only 1,
+      // and (4) the added item from the userDocs array is the user's uid.
+      function addUidToUserDocs() {
+        let onlyUserDocsUpdated = request.resource.data.keys().hasOnly('userDocs');
+        let oneItemRemoved = request.resource.data.users.size() == resource.data.users.size() + 1;
+        let addedUid = request.resource.data.users.removeAll(resource.data.users)[0] == request.auth.uid;
```

Comment:
this is a little hard to understand, consider something like `let removedUid = newData.removeAll(oldData)[0] = request.auth.uid;` where `oldData = resource.data.users` and `newData = request.resource.data.users`